pkg/bench/multipart_upload.go

```diff
@@ -0,0 +1,178 @@
+package bench
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"net/http"
+	"time"
+
+	"github.com/minio/minio-go/v7"
+	"golang.org/x/sync/errgroup"
+)
+
+// MultipartUpload benchmarks multipart upload speed.
+type MultipartUpload struct {
+	Common
+
+	PartsNumber      int
+	PartsConcurrency int
+}
+
+// Prepare for the benchmark run
+func (g *MultipartUpload) Prepare(ctx context.Context) error {
+	if err := g.createEmptyBucket(ctx); err != nil {
+		return err
+	}
+	return nil
+}
+
+// Start will execute the main benchmark.
+// Operations should begin executing when the start channel is closed.
+func (g *MultipartUpload) Start(ctx context.Context, wait chan struct{}) error {
+	eg, ctx := errgroup.WithContext(ctx)
+	c := g.Collector
+	if g.AutoTermDur > 0 {
+		ctx = c.AutoTerm(ctx, http.MethodPut, g.AutoTermScale, autoTermCheck, autoTermSamples, g.AutoTermDur)
+	}
+
+	for i := 0; i < g.Concurrency; i++ {
+		thread := uint16(i)
+		eg.Go(func() error {
+			<-wait
+
+			for {
+				select {
+				case <-ctx.Done():
+					return nil
+				default:
+				}
```

Comment:
As in your other code...

```suggestion
			for ctx.Err() == nil {
```

(you would need a return at the end, though)

Reply:
Done.